feat(navbar-tabs): Add tabs to the navbar

felipebrunetti94 opened the pull request:
![Gravação de tela de 07-11-2019 16_28_38](https://user-images.githubusercontent.com/47053687/68420763-c8d18180-017b-11ea-99ee-50572bb0a838.gif)

**Component**:

<!-- Navbar' -->

**Description**:
- Alter Navbar.component.js to include tabs
- Alter navbar.js to include tabs

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #116

<!-- If you want to refer to an issue while not closing it, use:

See: #116 

-->

felipebrunetti94 committed:
feat(navbar-tabs): Add tabs to the navbar

- Alter Navbar.component.js to include tabs
- Alter navbar.js to include tabs

Fix #116

felipebrunetti94 committed:
fix indentation

negreirosleo commented on src/lib/components/navbar/Navbar.component.js:
```diff
@@ -120,6 +148,13 @@ function NavBar({
         </NavbarMenuItem>
       </NavbarMenu>
+      <NavbarTabs>
+        {tabs.length > 0 &&
```
you can remove the `tab.length > 0` check here, since the default value is an empty array you can just leave the `.map()`.

negreirosleo commented on src/lib/components/navbar/Navbar.component.js:
```diff
@@ -120,6 +148,13 @@ function NavBar({
+        }
+      </NavbarTabs>
       <NavbarMenu>
         {languages.length > 0 && (
```
You also might do the same here :ok_woman:

carlito-scality replied:
I don't agree, it is not the same case here

negreirosleo replied:
oh, i'm sorry didnt saw the rest of the code, I assumed it was the same case

carlito-scality commented on src/lib/components/navbar/Navbar.component.js:
```diff
@@ -48,6 +55,26 @@ const NavbarMenu = styled.div`
+const NavbarTabs = styled.div`
+flex: 1;
+  display: flex;
+  justify-content: center;
```
It would be nicer to align tabs to left side => `justify-content: flex-start`

felipebrunetti94 replied:
Sure, will update, right now  :D

carlito-scality commented on src/lib/components/navbar/Navbar.component.js:
```diff
@@ -48,6 +55,26 @@ const NavbarMenu = styled.div`
+  align-items: center;
+`;
+
+const TabItems = styled.div`
```
You should manage the "selected" case, ie when a tab is `selected`, add a `border-bottom`

felipebrunetti94 replied:
Huumm... I'm using a border-bottom as hover, should i change ?

carlito-scality replied:
Yes you should keep it when hover as well ;)

carlito-scality commented on src/lib/components/navbar/Navbar.component.js:
```diff
@@ -48,6 +55,26 @@ const NavbarMenu = styled.div`
+  margin: 0 1vh;
+  :hover{
+    height: 100%;
+    border-bottom: 1vh solid black;
```
The border-bottom should be inside the navbar ;) like below by adding `box-sizing: border-box;`
![image](https://user-images.githubusercontent.com/47394132/68412437-7f3d6280-018c-11ea-8045-175618ccf4c1.png)

and make sure that all the item are aligned vertically.

carlito-scality replied:
Regarding color, please use `color: ${mergeTheme(props.theme, defaultTheme).primary};` for the label and the border  to make the component themeable

felipebrunetti94 replied:
okay :+1:

carlito-scality commented on src/lib/components/navbar/Navbar.component.js:
```diff
@@ -48,6 +55,26 @@ const NavbarMenu = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;  
+  margin: 0 1vh;
```
please use our predefined padding from `theme.js` to be consistent.
```

import {padding} from "../../style/theme"; 
```

felipebrunetti94 replied:
okay, didn't see this. I think creating a CONTRIBUTE, may be helpful in the future.

carlito-scality commented on src/lib/components/navbar/Navbar.component.js:
```diff
@@ -23,9 +28,11 @@ export type Props = {
   logo?: Node,
-  languages?: Items
+  languages?: Items,
+  tabs?:Tab,
```
It should be `tabs?: Array<Tab>`

felipebrunetti94 commented:
@carlito-scality , just one question the 
`
function NavBar({
  onToggleClick,
  toggleVisible,
  productName,
  applications,
  help,
  user,
  logo,
  tabs = [],
  languages = [],
  ...rest
}: Props)
` 
why do you use `...rest `?

carlito-scality commented:
« ...rest » allows you add more attributes if you want to (for example: id...)

negreirosleo commented:
@felipebrunetti94 Just a little tip, when creating a front-end feature its nice to have a video(in this case a gif) or an image of the component working so its easier to do the review :)

felipebrunetti94 committed:
feat(navbar-tabs): Add tabs to the navbar

    - Fix padding to use default
    - Fix color to use custom theme
    - Fix selected case

    Fix #116

felipebrunetti94 commented:
@carlito-scality , just done with the corrections.

carlito-scality commented on src/lib/components/navbar/Navbar.component.js:
```diff
@@ -8,12 +8,18 @@ import Button from "../button/Button.component";
 import * as defaultTheme from "../../style/theme";
 import { mergeTheme } from "../../utils";
 import type { Item } from "../dropdown/Dropdown.component";
+import {padding} from "../../style/theme"; 
```
You have already imported "../../style/theme" at the line 8

carlito-scality commented on src/lib/components/navbar/Navbar.component.js:
```diff
@@ -48,6 +56,43 @@ const NavbarMenu = styled.div`
+  align-items: center;  
+  padding: ${defaultTheme.padding.base};
+  :hover{
+    border-bottom: 1vh solid black;
```
it should be `mergeTheme(props.theme, defaultTheme).primary` instead of `black` and `2px ` instead of  `1vh`

carlito-scality commented on src/lib/components/navbar/Navbar.component.js:
```diff
@@ -48,6 +56,43 @@ const NavbarMenu = styled.div`
+      props.selected &&
+      css`
+        {
+          border-bottom: 1vh solid black;
```
Idem, it should be `mergeTheme(props.theme, defaultTheme).primary` instead of `black` and `2px ` instead of  `1vh`

carlito-scality commented on stories/index.js:
```diff
@@ -19,3 +19,4 @@ import "./modal";
 import "./tooltip";
 import "./multiselect";
 import "./chips";
+import "./progressbar";
```
You should not commit this in this PR ;)

felipebrunetti94 replied:
Yeah, I opened a new branch, but this one for some reason saved here....

carlito-scality commented on stories/navbar.js:
```diff
@@ -19,6 +19,29 @@ const user = {
+const tabs = [
+  {
+    selected: true,
+    title: "Did ",
```
- All the `title` should be more relevant like in the screenshot below even if they are funny :)
- onClick should be ```action( `${title}` clicked")```
- You should add an example to cover this case:
![image](https://user-images.githubusercontent.com/47394132/68472467-e86eb580-0220-11ea-8174-bf0945df0ed4.png)

carlito-scality commented on src/lib/components/navbar/Navbar.component.js:
```diff
@@ -48,6 +56,43 @@ const NavbarMenu = styled.div`
 `;
 
+const NavbarTabs = styled.div`
+flex: 1;
```
indentation here

carlito-scality commented on src/lib/components/navbar/Navbar.component.js:
```diff
@@ -48,6 +56,43 @@ const NavbarMenu = styled.div`
+`;
+
+const TabItems = styled.div`
+box-sizing: border-box;
```
indentation here

carlito-scality reviewed (changes requested):
Sorry, I still have some minor requests.
All the titles should be align vertically in center of the navbar. Indeed, the title of the selected tab or the title of hover tab should not be higher the rest of tab titles

felipebrunetti94 committed:
fix(navbar-tabs): Fix the last commit changes

- change 1vh to 2px
- change color to themecolor
- change onClick to onClick => act()

felipebrunetti94 commented:
@carlito-scality , done with changes, if you think there is still room for improvement, please tell me. Thank you :D

carlito-scality reviewed (approved):
@felipebrunetti94 LGTM. I added 1 commit to fine-tune styles and updated snapshot tests. Let's merge it. Thanks
I created https://github.com/scality/core-ui/pull/126 to merge your PR since we have our internal process.

carlito-scality commented:
merged in https://github.com/scality/core-ui/pull/126
